src/pages/components/ParticipantsForm.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from 'react';
+
+export default function ParticipantsForm({post}) {
+    const nameCount = 4; // default number of name fields
+    const [inputFields, setInputFields] = useState(
+        Array(nameCount).fill().map((_, i) => ({index: i, name: ''}))
+      )
+    
+    const handleFormChange = (index, event) => {
+        let data = [...inputFields];
+        data[index][event.target.name] = event.target.value;
+        setInputFields(data);
+    }
+    
+    const addField = () => {
+        const newField = {name: ''};
+        setInputFields([...inputFields, newField])
+    }
+    
+    const submit = (e) => {
+        e.preventDefault(); // prevents page refresh
+        console.log(inputFields);
+        post(inputFields.map(({index, name}) => name))
+    }
+
+    return (
+    <div>
+        <p className='recvField'>
+        <div className="App">
```

Comment:
Warning: <div> cannot appear as a descendant of <p>.

Good explanation from [here](https://stackoverflow.com/questions/41928567/div-cannot-appear-as-a-descendant-of-p)
"According to this [document](https://www.w3.org/TR/html401/struct/text.html#h-9.3.1), a <p></p> tag can only contain inline elements. That means putting a <div></div> tag inside it should be improper, since the div tag is a block element. Improper nesting might cause glitches like rendering [extra tags](https://stackoverflow.com/questions/10763780/putting-div-inside-p-is-adding-an-extra-p), which can affect your javascript and css."

Reply:
removed paragraph